src/imageops/affine.rs

```diff
@@ -97,3 +97,115 @@ pub fn flip_vertical<I: GenericImage + 'static>(image:  &I)
 
     out
 }
+
+#[cfg(test)]
+mod test {
+    use super::{rotate90,rotate180,rotate270,flip_horizontal,flip_vertical};
+    use buffer::{ImageBuffer,Pixel,GrayImage};
+    use image::{GenericImage};
+
+    macro_rules! assert_pixels_eq {
+        ($actual:expr, $expected:expr) => ({
+            let actual_dim = $actual.dimensions();
+            let expected_dim = $expected.dimensions();
+
+            if actual_dim != expected_dim {
+                panic!("dimensions do not match. \
```

Comment:
"this" in my question refering to `panic!`. Answering my own question from http://stackoverflow.com/questions/24718256/higher-order-macros

> It works fine when the nested macro has no argument itself, because there is no nested $... non-terminals and so the expansion phase doesn't ever hit the definition problem described above.